rest/server_context.go

```diff
@@ -797,6 +797,21 @@ func (sc *ServerContext) initEventHandlers(dbcontext *db.DatabaseContext, config
 			return pkgerrors.Wrapf(err, "Error calling base.JSONUnmarshal() in initEventHandlers")
 		}
 
+		// Load Webhook Filter Function.
+		for _, conf := range append(eventHandlers.DBStateChanged, eventHandlers.DocumentChanged...) {
```

Comment:
The append threw me for a second until I realised what was going on.

What do you think about this approach instead? Then the same loop can be used for both js loading, and registration, and is quite a bit easier to add new handlers when we need to.

```go
eventHandlersByType := map[db.EventType][]*EventConfig{
	db.DocumentChange: eventHandlers.DocumentChanged,
	db.DBStateChange:  eventHandlers.DBStateChanged,
}

for eventType, handlers := range eventHandlersByType {
	// Load external webhook filter function
	for _, conf := range handlers {
		filter, err := loadJavaScript(conf.Filter)
		if err != nil {
			return &JavaScriptLoadError{
				JSLoadType: WebhookFilter,
				Path:       conf.Filter,
				Err:        err,
			}
		}
		conf.Filter = filter
	}

	// Register event handlers
	if err = sc.processEventHandlersForEvent(handlers, eventType, dbcontext); err != nil {
		return err
	}
}
```

Reply:
Sure this makes sense. Change are included in the latest revision.